gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManager.java

```diff
@@ -0,0 +1,430 @@
+    String flowGroup = killFlowEvent.getFlowGroup();
+    String flowName = killFlowEvent.getFlowName();
+    if (killFlowEvent.getFlowExecutionId() != null) {
+      long flowExecutionId = killFlowEvent.getFlowExecutionId();
+      log.info("Received kill request for flow ({}, {}, {})", flowGroup, flowName, flowExecutionId);
+      // todo - should this be added to the dac action store instead?
+      this.dagTaskStream.addDagAction(new DagActionStore.DagAction(flowGroup, flowName, String.valueOf(flowExecutionId), DagActionStore.FlowActionType.KILL));
+
+    } else {
+      List<Long> flowExecutionIds = getJobStatusRetriever().getLatestExecutionIdsForFlow(flowName, flowGroup, 10);
+      log.info("Found {} flows to cancel.", flowExecutionIds.size());
+      for (long flowExecutionId : flowExecutionIds) {
+        handleKillFlowEvent(new KillFlowEvent(flowGroup, flowName, flowExecutionId));
+      }
+    }
+  }
+
+  @Subscribe
+  public void handleResumeFlowEvent(ResumeFlowEvent resumeFlowEvent) {
+    if (!this.dagProcessingEngineEnabled) {
+      log.info("Not handling resume flow event yet.");
+      return;
+    }
+    String flowGroup = resumeFlowEvent.getFlowGroup();
+    String flowName = resumeFlowEvent.getFlowName();
+    String flowExecutionId = String.valueOf(resumeFlowEvent.getFlowExecutionId());
+    log.info("Received resume request for flow ({}, {}, {})", flowGroup, flowName, flowExecutionId);
+
+    // todo - persist dag action instead of directly adding to the stream
+    this.dagTaskStream.addDagAction(new DagActionStore.DagAction(flowGroup, flowName, String.valueOf(flowExecutionId), DagActionStore.FlowActionType.RESUME));
+  }
+
+  @Subscribe
+  // todo - this does not sla kill like DagManager does, that functionality is to move somewhere else
+  public void handleJobStatusEvent(JobStatusEvent jobStatusEvent) {
+    if (!this.dagProcessingEngineEnabled) {
+      return;
+    }
+    Map<String, Set<Dag.DagNode<JobExecutionPlan>>> nextSubmitted = Maps.newHashMap();
+    List<Dag.DagNode<JobExecutionPlan>> nodesToCleanUp = Lists.newArrayList();
+
+    ExecutionStatus executionStatus = jobStatusEvent.getStatus();
+    JobStatus jobStatus = jobStatusEvent.getJobStatus();
+
+    String dagNodeId = DagManagerUtils.generateDagNodeId(jobStatusEvent);
+    Dag.DagNode<JobExecutionPlan> dagNode = this.dagManagementStateStore.getDagNode(dagNodeId);
+
+    JobExecutionPlan jobExecutionPlan = DagManagerUtils.getJobExecutionPlan(dagNode);
+
+    try {
+        switch (executionStatus) {
+          case COMPLETE:
+            jobExecutionPlan.setExecutionStatus(COMPLETE);
+            nextSubmitted.putAll(onJobFinish(dagNode));
+            nodesToCleanUp.add(dagNode);
+            break;
+          case FAILED:
+            jobExecutionPlan.setExecutionStatus(FAILED);
+            nextSubmitted.putAll(onJobFinish(dagNode));
+            nodesToCleanUp.add(dagNode);
+            break;
```

Comment:
if using `switch`, let's use `case` fall-through to our advantage... otherwise use `if/else` to capture that there are truly only 3 alternatives.
e.g.
```
switch (execStatus) {
    case COMPLETE:
    case FAILED:
    case CANCELLED:
      jobExecPlan.setExecStatus(execStatus);
      nextSub.putAll(onFin(...));
      nodesToClean.add(...);
      break
    case PENDING:
    case PENDING_RETRY:
      jobExecPlan.setExecStatus(execStatus);
       break;
    default:
      ...
}
```